tests/test_base_config.py

```diff
@@ -618,14 +637,34 @@ def test_copy_with_interpolation(
         cp[interpolated_key] = "XXX"
         assert cp[interpolated_key] == cp[interpolating_key]
 
-    def test_list_copy_is_shallow(self, copy_method: Any) -> None:
+    def test_list_shallow_copy_is_deepcopy(self, copy_method: Any) -> None:
```

Comment:
Should there be a similar test for dicts?

Reply:
Yeah, was a bit lazy.
I think we should probably reorganize all the copy tests and put them in a single test file. it's quiet a mess now.